File: palindrome/go/palindrome.go
package main

import (
	"strings"
	"unicode"
)

func isPalindrome(s string) bool {
	// 1. Convert to lowercase and remove non-letter characters
	var cleaned strings.Builder
	for _, r := range s {
		if unicode.IsLetter(r) {
			cleaned.WriteRune(unicode.ToLower(r))
		}
	}
	cleanedStr := cleaned.String()
	// 2. Reverse the cleaned string
	reversed := reverseString(cleanedStr)
	// 3. Compare cleaned string with its reverse
	return cleanedStr == reversed
}

File: languages/typescript/solutions/palindrome/palindrome.ts
export function isPalindrome(str: string): boolean {
    // 1. Convert the string to lowercase
    const lowerStr = str.toLowerCase();
    // 2. Remove all non-alphanumeric characters
    const alphanumericStr = lowerStr.replace(/[^a-z0-9]/g, '');
    // 3. Reverse the string
    const reversedStr = reverseString(alphanumericStr);
    // 4. Compare the original and reversed strings
    return alphanumericStr === reversedStr;
}

function reverseString (str: string): string {
    return str.split('').reverse().join('');
}